src/daft-catalog/src/session.rs

```diff
@@ -0,0 +1,51 @@
+use daft_logical_plan::LogicalPlanBuilder;
+
+use crate::{errors::Result, identifier::Identifier, DaftCatalog};
+
+/// Temporary rename while refactoring.
+type Metastore = DaftCatalog;
+
+/// Session holds all state for query planning and execution (e.g. connection).
+pub struct Session {
```

Comment:
I was thinking a bit about this while implementing the `daft-context` crate. Should `Session` be part of the catalog crate, or should it be it's own thing? The way I see it is that the Session holds a catalog, and is independent of the catalog so it might make sense to pull it into it's own crate.

Reply:
Yes this was one of my questions as well, and I made it its own crate to begin with. In both PartiQL and DataFusion the session is in the catalog so I just put it in here for a smaller diff. I agree and will move it back to its own.